platforms/core-runtime/messaging/src/main/java/org/gradle/internal/dispatch/ProxyDispatchAdapter.java

```diff
@@ -69,26 +85,25 @@ private DispatchingInvocationHandler(Class<?> type, Dispatch<? super MethodInvoc
 
         @Override
         public Object invoke(Object target, Method method, Object[] parameters) throws Throwable {
-            if (method.getName().equals("equals")) {
-                Object parameter = parameters[0];
-                if (parameter == null || !Proxy.isProxyClass(parameter.getClass())) {
-                    return false;
-                }
-                Object handler = Proxy.getInvocationHandler(parameter);
-                if (!DispatchingInvocationHandler.class.isInstance(handler)) {
-                    return false;
-                }
+            switch (method.getName()) {
+                case "equals":
+                    Object parameter = parameters[0];
+                    if (parameter == null || !Proxy.isProxyClass(parameter.getClass())) {
+                        return false;
+                    }
+                    Object handler = Proxy.getInvocationHandler(parameter);
+                    if (!(handler instanceof DispatchingInvocationHandler)) {
+                        return false;
+                    }
 
-                DispatchingInvocationHandler otherHandler = (DispatchingInvocationHandler) handler;
-                return otherHandler.type.equals(type) && otherHandler.dispatch == dispatch;
+                    DispatchingInvocationHandler otherHandler = (DispatchingInvocationHandler) handler;
+                    return otherHandler.type.equals(type) && otherHandler.dispatch == dispatch;
+                case "hashCode":
+                    return dispatch.hashCode();
+                case "toString":
+                    return type.getSimpleName() + " broadcast";
             }
 
-            if (method.getName().equals("hashCode")) {
-                return dispatch.hashCode();
-            }
-            if (method.getName().equals("toString")) {
-                return type.getSimpleName() + " broadcast";
-            }
             dispatch.dispatch(new MethodInvocation(method, parameters));
```

Comment:
💅 This can be folded under `default:` switch branch.

Reply:
Thanks, @mlopatkin. Done in 3e76f0fa9bd50382745330cde83c9257c3eed289